src/main/kotlin/io/libp2p/pubsub/gossip/GossipScore.kt

```diff
@@ -117,6 +118,7 @@ class GossipScore(
         val ips = mutableSetOf<String>()
         var connectedTimeMillis: Long = 0
         var disconnectedTimeMillis: Long = 0
+        val cachedScore: AtomicReference<Double> = AtomicReference(0.0)
```

Comment:
- `AtomicDouble` could be used here 
- All the gossip structures are not thread safe by design (they are accessed from a single thread). Accessing `peerScores[peerId]` would most likely work thought is still not thread safe. With respect to this I'm not sure that using `Atomic*` makes too much sense here. Probably a simple `var cachedScore: Double` would be enough?

Reply:
Makes sense - looks like we should just make `peerScores` a `ConcurrentHashMap`, and use a plain `Double` for the cached value.  WDYT?

Reply:
Yeah makes sense. I'd also mark it with `@Volatile` to be safe 
Alternatively we may consider a dedicated `ConcurrentHashMap` for the score cache